### ISSUE TITLE
skipUntil unsubscription potential bug

### ISSUE THREAD

**Issue by staltz** [ReactiveX/RxJS#577]
I'm writing unsubscription tests for skipUntil, and I expect this to be true
```js
var e1 =   hot('--a--b--c--d--e--|');
var e1subs =   '^                !'; // Notice this
var skip = hot('-');
var skipSubs = '^';
var expected = '-';
```
but in reality RxJS Next behaves like this
```js
var e1 =   hot('--a--b--c--d--e--|');
var e1subs =   '^'; // Notice this
var skip = hot('-');
var skipSubs = '^';
var expected = '-';
```

RxJS 4, on the other hand, unsubscribes from the source when the source completes, see this JSBin: http://jsbin.com/vumuxurabe/1/edit?js,console.

Should I consider this a bug in RxJS Next?

**Comment by kwonoj** [comment]
I feel same, subscription need to be unsubscribed.

**Comment by blesh** [comment]
@staltz ... I'm going through issues and making sure what needs to be closed is closed... is this issue resolved now with #586?

**Comment by staltz** [comment]
Yes, let's close.
